### PR TITLE
Code quality improvements based on core feedback

### PR DESCRIPTION
**Pull request by iMicknl**

### CODE REVIEW

File: custom_components/tahoma/__init__.py [hunk 1]
```diff
@@ -111,47 +111,55 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
...
+        _LOGGER.error("Server is down for maintenance.")
         raise ConfigEntryNotReady from exception
     except Exception as exception:  # pylint: disable=broad-except
         _LOGGER.exception(exception)
         return False
 
-    update_interval = entry.options.get(CONF_UPDATE_INTERVAL, DEFAULT_UPDATE_INTERVAL)
+    update_interval = timedelta(
+        seconds=entry.options.get(CONF_UPDATE_INTERVAL, DEFAULT_UPDATE_INTERVAL)
+    )
```
Comment (tetienne) [inline custom_components/tahoma/__init__.py:131]: Why change this?

File: custom_components/tahoma/config_flow.py [hunk 2]
```diff
@@ -130,16 +126,17 @@ async def async_step_init(self, user_input=None):
...
         return self.async_show_form(
             step_id="update_interval",
             data_schema=vol.Schema(
                 {
                     vol.Required(
                         CONF_UPDATE_INTERVAL,
-                        default=self.options.get(CONF_UPDATE_INTERVAL),
+                        default=self.config_entry.options.get(
+                            CONF_UPDATE_INTERVAL, DEFAULT_UPDATE_INTERVAL
+                        ),
```
Comment (tetienne) [inline custom_components/tahoma/config_flow.py:139]: What’s the difference?

File: custom_components/tahoma/__init__.py (hunk 1 above)
Comment (iMicknl) [reply custom_components/tahoma/__init__.py:131]: I had to move the following debug from our __init__ method in the Data Update Coordinator. Moving up the timedelta made it easier to reuse this variable for the DataUpdateCoordinator and debug logging.

```python
_LOGGER.debug(
        "Initialized DataUpdateCoordinator with %s interval.", str(update_interval)
    )
```

File: custom_components/tahoma/config_flow.py (hunk 2 above)
Comment (iMicknl) [reply custom_components/tahoma/config_flow.py:139]: Fallback to `DEFAULT_UPDATE_INTERVAL`, instead of setting that in the constructor.

**Review by tetienne — APPROVED** [review]

**Pull request merged**